make_profiler/report_export.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/python3
+
+from datetime import datetime
+DATE_FORMAT = "%Y-%m-%dT%H:%M:%SZ"
+
+
+def escape_json_string(s):
+    s=str(s)
+    s = s.replace("\"", "")
+    s = s.replace("'","")
+    s = s.replace("\\","\\\\") # single \ to double \\
+    return s
+
+
+def export_report(performance,docs):
+    strOutputFile = 'report.json'
+    fo = open(strOutputFile, 'w', encoding="utf-8")
+    n_in_progress=0
+    n_failed=0
+    n_total = 0
+    oldest_completed_target = ''
+    fo.write('{\n')
+    fo.write('    "status":[\n')
+    for key in performance:
+        rec=performance[key]
+        n_total += 1
+        if rec["failed"] == True:
+            event_type = "failed"
+            n_failed += 1
+        elif rec["done"] == True:
+            event_type = "completed"
+        elif rec["running"] == True:
+            event_type = "started"
+            n_in_progress += 1
+        else:
+            event_type = "??unknown??"
+
+        #event_time = datetime.fromtimestamp(int(rec['start_current'])).strftime(DATE_FORMAT)
+        if 'finish_prev' in rec:
+            last_event_time = datetime.fromtimestamp(int(rec['finish_prev'])).strftime(DATE_FORMAT)
+        else:
+            last_event_time = ''
+
+        if 'start_current' in rec:
+            event_time = datetime.fromtimestamp(int(rec['finish_current'])).strftime(DATE_FORMAT)
+        else:
+            event_time = last_event_time
+
+        if event_type == "completed":
+            if oldest_completed_target == '':
+                oldest_completed_target = event_time
+            if event_time<oldest_completed_target:
+                oldest_completed_target = event_time
+
+        descr = docs.get(key, '')
+        event_duration=rec.get("timing_sec",0)
+        if n_total > 1:
+            fo.write(',\n')
+        fo.write('        {"eventN":"'+escape_json_string(key) +'",\n')
```

Comment:
You can use string formatting for better code readability, like this:
```python
fo.write('        {{"eventN":"{0}",\n'.format(escape_json_string(key)))
```
Or even better:
```python
fo.write(f'        {{"eventN":"{escape_json_string(key)}",\n')
```
But for the second option you will need to require Python version >= 3.6
You can also substitute spaces by using `" " * 8` but I'm not sure it adds any simplicity to the code:
```python
fo.write(f'{" " * 8}{{"eventN":"{escape_json_string(key)}",\n')
```

Reply:
@eokulik json should be passed as json and not manipulated as text, unless you're developer of json parsing library.

Reply:
https://github.com/konturio/make-profiler/pull/11/commits/82ee16373f02788023fe7bfa0619fa444f69c297